swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/SharedAccess.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  SharedAccess.swift
+//  (c) 2023 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import Foundation
+
+public struct SharedAccess {
+  public static var baseFolderURL: URL {
+    guard let url = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: AppInfoPlistConstants.appGroupId) else {
+      fatalError("Shared folder unavailable")
+    }
+    return url
+  }
+
+  public static var cacheFolderURL: URL? {
+    let url = baseFolderURL.appendingPathComponent("Library").appendingPathComponent("Caches")
+    guard ensureDirectoryExists(at: url.path) else {
+      return nil
+    }
+    return url
+  }
+
+  public static var logFolderURL: URL? {
+    if let url = cacheFolderURL?.appendingPathComponent("logs") {
+      guard ensureDirectoryExists(at: url.path) else {
+        return nil
+      }
+      return url
+    }
+    NSLog("Can't access cacheFolderURL to create logFolderURL")
+    return nil
+  }
+
+  public static var connlibLogFolderURL: URL? {
+    if let url = logFolderURL?.appendingPathComponent("connlib") {
+      guard ensureDirectoryExists(at: url.path) else {
+        return nil
+      }
+      return url
+    }
+    NSLog("Can't access logFolderURL to create connlibLogFolderURL")
+    return nil
+  }
+
```

Comment:
(nit) Aren't these considered `URI`s not `URL`s?

Reply:
These *are* URLs in Apple parlance -- we're using the `FileManager.containerURL` API to get it.

URIs are a superset of URLs, so these are URIs too.